File: sinq/src/test/java/models/Student_.java
package models;

import javax.persistence.metamodel.SetAttribute;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

/**
 * User: YaFengLi
 * Date: 12-12-14
 * Time: 下午5:36
 */
@StaticMetamodel(Student.class)
public class Student_ {
    public static volatile SingularAttribute<Student, Long> id;
    public static volatile SingularAttribute<Student, String> name;
    public static volatile SingularAttribute<Student, String> address;
    public static volatile SingularAttribute<Student, Integer> age;
    public static volatile SingularAttribute<Student, Teacher> teacher;
    public static volatile SetAttribute<Student, Book> books;
}

File: sinq/src/test/scala/models/Student.scala
package models

import javax.persistence._

import scala.beans.BeanProperty


@Entity
@Table(name = "t_student")
case class Student(@BeanProperty var name: String, @BeanProperty var age: Int, @BeanProperty var address: String) {
  @Id
  @GeneratedValue(strategy = GenerationType.TABLE, generator = "seq_t_student")
  @TableGenerator(name = "seq_t_student", table = "seq_t_student", allocationSize = 1)
  @BeanProperty
  var id: Long = _

  @ManyToOne(fetch = FetchType.EAGER, optional = false)
  @JoinColumn(name = "teacher_id")
  @BeanProperty
  var teacher: Teacher = _

  def this() = this(null, -1, null)

  def this(name: String) = this(name, -1, null)

  def this(name: String, age: Int) = this(name, age, null)

  def this(name: String, age: Int, address: String, teacher: Teacher) = {
    this(name, age, address)
    this.teacher = teacher
  }

  override def toString = id + ":" + name + ":" + age + ":" + address
}